tests/analysis_tests.py

```diff
@@ -45,6 +45,17 @@ def test_segment_rowsize_raises(self):
         with self.assertRaises(ValueError):
             segment(x, tol, rowsize)
 
+    def test_segments_datetime(self):
+        x = [
+            np.datetime64("2023-01-01"),
+            np.datetime64("2023-01-02"),
+            np.datetime64("2023-01-03"),
+            np.datetime64("2023-02-01"),
+            np.datetime64("2010-02-01"),
```

Comment:
I think there's an issue here. Did you mean "2023-02-02"?

Reply:
Let's put a test with hour intervals. We want the tolerance to be the maximum difference allowed so that `tol = np.timedelta64(1,"H")` will split the GDP hourly ragged array, correct?

Reply:
Yes, that was a typo. I'm not sure I understand the difference with the test you want to do (apart from 1h instead of 1d).